manager/util/tokenrequest.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"k8s.io/client-go/rest"
+
+	"github.com/cert-manager/csi-lib/manager"
+	"github.com/cert-manager/csi-lib/metadata"
+)
+
+// ClientForMetadataTokenRequestEmptyAud returns a
+// manager.ClientForMetadataFunc that returns a cert-manager rest client whose
+// authentication is built using the passed empty audience ("") token request
+// in the metadata VolumeContext. The resulting cert-manager client is
+// authenticated against the Kubernetes API server using the mounting Pod's
+// ServiceAccount.
+//
+// Intended to be used as a manager ClientForMetadata so that created
+// CertificateRequests will have UserInfo fields of the mounting Pods
+// ServiceAccount.
+//
+// Drivers using this function _must_ have the empty audience tokenRequest
+// defined on the CSIDriver manifest definition, along with setting
+// requiresRepublish to true:
+//
+// tokenRequests:
+//   - audience: ""
+//     expirationSeconds: 3600
+// requiresRepublish: true
+//
+// restConfig must contain the Kubernetes API server Host, and a valid
+// TLSClientConfig.
+func ClientForMetadataTokenRequestEmptyAud(restConfig *rest.Config) manager.ClientForMetadataFunc {
```

Comment:
Suggestion: indicate in the function name that it's cert-manager client.

Reply:
I'm inclined to say no. All consumer facing clients are cert-manager clients, [see](https://github.com/cert-manager/csi-lib/blob/dd8ae5d66f5301940c865704f3e91992ab3bc4d6/manager/manager.go#L58).

I think extra context would be needed if it _wasn't_ a cert-manager client.

Reply:
Ah ok, that makes sense. I was thinking about it like about a controller where the default client think is a core kube client 👍🏼